legend-engine-pure-code-compiled-core/src/main/resources/core/pure/router/routing/router_routing.pure

```diff
@@ -162,22 +162,38 @@ function <<access.private>> meta::pure::router::routing::routeValueSpecification
    );
 }
 
+function meta::pure::router::routing::prevalFunctionExpressionIfRequired(fe:FunctionExpression[1], inScopeVars:Map<String, List<Any>>[1], extensions:meta::pure::extension::Extension[*], debug:DebugContext[1]) : FunctionExpression[1]
+{
+    $fe.func->match(
+      [
+        f : FunctionDefinition<Any>[1] | if($f->hasStereotype('NormalizeRequiredFunction', functionType), 
+                                        | $fe->meta::pure::router::preeval::preval($inScopeVars->putAll($f->openVariableValues()), $extensions, noDebug()),
+                                        | $fe),
+        a:Any[*] | $fe;
+      ]
+    )->cast(@FunctionExpression);
+}
+
 function meta::pure::router::routing::routeFunctionExpression(fe:FunctionExpression[1], state:RoutingState[1], executionContext:ExecutionContext[1], vars:Map<VariableExpression, ValueSpecification>[1], inScopeVars:Map<String, List<Any>>[1], extensions:meta::pure::extension::Extension[*], debug:DebugContext[1]):RoutingState[1]
 {
+   
+
    $fe.func->match([
      p:Property<Nil,Any|*>[1] | $state.routingStrategy.processProperty($p, $fe, $state, $executionContext, $vars, $inScopeVars, $extensions, $debug);,
      f:Function<Any>[1]       | print(if($debug.debug,|$debug.space+if($f->instanceOf(NativeFunction),|'NativeFunction',|'FunctionDefinition')+' : '+$f->asString()+'\n',|''));
                                 let specializedFunctionProcessor = specializedFunctionExpressionRouterProcessor($extensions)->filter(e|$e.first->eval($fe)).second;
                                 if($specializedFunctionProcessor->isNotEmpty(),
                                   | assert($specializedFunctionProcessor->size() == 1, | 'Only one match is authorized for specializedFunctionProcessor');
                                     $specializedFunctionProcessor->toOne()->evaluate([list($f), list($fe), list($state), list($executionContext), list($vars), list($inScopeVars), list($extensions), list($debug)])->toOne()->cast(@RoutingState);,
-                                  | $f->routeFunctionExpressionFunctionDefinition($fe, $state, $executionContext, $vars, $inScopeVars->putAll($f->openVariableValues()), $extensions, $debug));
+                                  | let newFe = prevalFunctionExpressionIfRequired($fe,$inScopeVars, $extensions, $debug);
```

Comment:
What's you view on doing preeval processing for all function whether specialized processor exists or not? I see 2 views : 
1. If we do it before we can centralize that logic call from one place.
2. If we do it after we say someone has specified specialized processor and they should have full control

Reply:
After discussing I think we want to go with 1 - have modified the change to process prior to specialized processors